processing/src/main/java/org/apache/druid/query/filter/NullFilter.java

```diff
@@ -0,0 +1,315 @@
+import org.apache.druid.segment.vector.VectorColumnSelectorFactory;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+public class NullFilter extends AbstractOptimizableDimFilter implements Filter
+{
+  public static NullFilter forColumn(String column)
+  {
+    return new NullFilter(column, null, null);
+  }
+
+  private final String column;
+  @Nullable
+  private final ExtractionFn extractionFn;
+  @Nullable
+  private final FilterTuning filterTuning;
+
+  @JsonCreator
+  public NullFilter(
+      @JsonProperty("column") String column,
+      @JsonProperty("extractionFn") @Nullable ExtractionFn extractionFn,
+      @JsonProperty("filterTuning") @Nullable FilterTuning filterTuning
+  )
+  {
+    if (column == null) {
+      throw InvalidInput.exception("Invalid null filter, column cannot be null");
+    }
+    this.column = column;
+    // remove once SQL planner no longer uses extractionFn
+    this.extractionFn = extractionFn;
+    this.filterTuning = filterTuning;
+  }
+
+  @JsonProperty
+  public String getColumn()
+  {
+    return column;
+  }
+
+  @Nullable
+  @JsonProperty
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  public ExtractionFn getExtractionFn()
+  {
+    return extractionFn;
+  }
+
+  @Nullable
+  @JsonProperty
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  public FilterTuning getFilterTuning()
+  {
+    return filterTuning;
+  }
+
+  @Override
+  public byte[] getCacheKey()
+  {
+    return new CacheKeyBuilder(DimFilterUtils.NULL_CACHE_ID)
+        .appendByte(DimFilterUtils.STRING_SEPARATOR)
+        .appendString(column)
+        .appendByte(DimFilterUtils.STRING_SEPARATOR)
+        .appendByteArray(extractionFn == null ? new byte[0] : extractionFn.getCacheKey())
+        .build();
+  }
+
+  @Override
+  public DimFilter optimize()
+  {
+    return this;
+  }
+
+  @Override
+  public Filter toFilter()
+  {
+    if (extractionFn == null) {
+      return this;
+    } else {
+      return new DimensionPredicateFilter(column, NullPredicateFactory.INSTANCE, extractionFn, filterTuning);
+    }
+  }
+
+  @Nullable
+  @Override
+  public RangeSet<String> getDimensionRangeSet(String dimension)
+  {
+    RangeSet<String> retSet = TreeRangeSet.create();
+    // Nulls are less than empty String in segments
+    retSet.add(Range.lessThan(""));
+    return retSet;
+  }
+
+  @Nullable
+  @Override
+  public BitmapColumnIndex getBitmapColumnIndex(ColumnIndexSelector selector)
+  {
+    if (!Filters.checkFilterTuningUseIndex(column, selector, filterTuning)) {
+      return null;
+    }
+    final ColumnIndexSupplier indexSupplier = selector.getIndexSupplier(column);
+    if (indexSupplier == null) {
+      return Filters.makeNullIndex(true, selector);
+    }
+    final NullValueIndex nullValueIndex = indexSupplier.as(NullValueIndex.class);
+    if (nullValueIndex == null) {
+      return null;
+    }
+    return nullValueIndex.forNull();
```

Comment:
I think i called it `forNull()` because the value set indexes have `forValue(...)` and `NullValueIndex` was originally split out of that. e.g. `BitmapColumnIndex  index = indexes.forValue(...)` ... `... indexes.forNull()` etc. Can change :shrug:

Reply:
renamed to `get`